core/src/main/scala/chisel3/experimental/package.scala

```diff
@@ -162,6 +162,35 @@ package object experimental {
     */
   trait NoChiselNamePrefix
 
+  /** Treat instances of this type as a named component, and rename using the Chisel compiler
+    * plugin.
+    *
+    * The compiler plugin by default automatically prefixes and names instances of `Data` and
+    * `Mem` recursively. This trait allows the plugin to treat arbitrary types as named
+    * components and prefix them (and objects constructed within them) when they otherwise
+    * would not be.
```

Comment:
```suggestion
  /** Generate prefixes from values of this type in the Chisel compiler plugin
    *
    * Users can mixin this trait to tell the Chisel compiler plugin to include the names of
    * vals of this type when generating prefixes for naming `Data` and `Mem` instances.
    * This is generally useful whenever creating a `class` that contains `Data`, `Mem`,
    * or `Module` instances but does not itself extend `Data` or `Module`.
```
This is a subtle but tricky thing to communicate. I think it's important to not call instances of this type "named components" because they themselves are not named, it's just that values of this type can affect the prefix of things that are named. It might be useful to include a link to the Chisel docs page on this (using the `@see` ScalaDoc tag).